hazelcast/src/main/java/com/hazelcast/map/impl/querycache/subscriber/DefaultQueryCache.java

```diff
@@ -343,7 +344,7 @@ public V get(Object key) {
         Set<QueryableEntry> query = indexes.query(predicate);
         if (query != null) {
             for (QueryableEntry entry : query) {
-                K key = (K) entry.getKey();
+                K key = (K) serializationService.toObject(entry.getKeyData());
```

Comment:
Minor: there is a shortcut `toObject()` method for doing the deserialization in this class and in `DefaultQueryCache`. Maybe that should be used to be consistent at all places.